openedx/features/course_experience/static/course_experience/js/spec/CourseHome_spec.js

```diff
@@ -57,13 +57,14 @@ describe('Course Home factory', () => {
 
     it('should send events to Segment and edX on initial load', () => {
       expect(window.analytics.track).toHaveBeenCalledWith('Promotion Viewed', segmentEventProperties);
-      expect(Logger.log).toHaveBeenCalledWith('edx.course.upgrade.hero.displayed', { courseRunKey: runKey });
+      expect(Logger.log).toHaveBeenCalledWith('edx.bi.course.upgrade.hero.displayed', { courseRunKey: runKey });
     });
 
     it('should send events to Segment and edX after clicking the upgrade button ', () => {
       $('.vc-message .btn-upgrade').click();
       expect(window.analytics.track).toHaveBeenCalledWith('Promotion Viewed', segmentEventProperties);
-      expect(Logger.log).toHaveBeenCalledWith('edx.course.upgrade.hero.clicked', { courseRunKey: runKey });
+      expect(Logger.log).toHaveBeenCalledWith('edx.bi.course.upgrade.hero.clicked', { courseRunKey: runKey });
+      expect(Logger.log).toHaveBeenCalledWith('edx.course.enrollment.upgrade.clicked', { location: 'hero' });
```

Comment:
@andy-armstrong:  Quality rules for JS seem really odd.  Here it complained when I didn't have spaces after/before '{' and '}' (e.g. "{location: 'hero'}", but in another location above, it complained when I tried adding them. 

The code where it complained:
```
expect(Logger.log).toHaveBeenCalledWith('edx.course.enrollment.upgrade.clicked',
    {location: 'date-sidebar'}
);
```
It also didn't complain about the following in the above line:
```
Logger.log('edx.course.enrollment.upgrade.clicked', {location: 'date-sidebar'});
```
What's the difference?  Different JS versions with different rules?  Other?